File: reactNativeApp/modules/slax-bridge/android/src/main/java/com/slax/bridge/SlaxBridgeModule.kt
package com.slax.bridge

import com.slax.bridge.api.NativeBridgeRegistry
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.Promise

class SlaxBridgeModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("SlaxBridge")

        AsyncFunction("invoke") { method: String, payload: Map<String, Any?>, promise: Promise ->
            val handler = NativeBridgeRegistry.handler
                ?: run {
                    promise.reject("ERR_NOT_INITIALIZED", "NativeBridge not initialized", null)
                    return@AsyncFunction
                }
            handler.invoke(method, payload) { result ->
                result.fold(
                    onSuccess = { promise.resolve(it) },
                    onFailure = { promise.reject("ERR_BRIDGE", it.message, it) }
                )
            }
        }
    }
}

File: reactNativeApp/modules/slax-bridge/ios/SlaxBridgeModule.swift
import ExpoModulesCore
import SlaxBridgeCore

public class SlaxBridgeModule: Module {
    public func definition() -> ModuleDefinition {
        Name("SlaxBridge")

        AsyncFunction("invoke") { (method: String, payload: [String: Any], promise: Promise) in
            guard let handler = SlaxBridgeRegistry.handler else {
                promise.reject("ERR_NOT_INITIALIZED", "SlaxBridge handler not registered")
                return
            }
            handler(method, payload) { result in
                switch result {
                case .success(let value):
                    promise.resolve(value)
                case .failure(let error):
                    promise.reject("ERR_BRIDGE", error.localizedDescription)
                }
            }
        }
    }
}
